nvflare/apis/impl/wf_comm_server.py

```diff
@@ -147,8 +148,15 @@ def handle_event(self, event_type: str, fl_ctx: FLContext):
             event_type (str): all event types, including AppEventType and EventType
             fl_ctx (FLContext): FLContext information with current event type
         """
-        if event_type == InfoCollector.EVENT_TYPE_GET_STATS:
+        if event_type == InfoCollector.EVENT_TYPE_SET_STATS:
```

Comment:
I'm not sure why you need two event types here.  Please explain.

Reply:
Originally many classes (ScatterAndGather, CrossSiteModelEval, ModelController etc.) handle the `InfoCollector.EVENT_TYPE_GET_STATS` which calls `GroupInfoCollector:add_info()`. However beforehand they all call the controller's `super().handle_event(event_type, fl_ctx)` which does `GroupInfoCollector:set_info()`.

Now that we are removing `handle_event()` from the controller (and keeping it in the communicator), the communicator's `set_info` still must be called first before `add_info` so it doesn't override the info, so handling the same event will no longer work. I added a `InfoCollector.EVENT_TYPE_SET_STATS` event first, and then the communicator fires `InfoCollector.EVENT_TYPE_GET_STATS` afterwards.

Reply:
The purpose of the GET_STATS event is simply to collect info from components that are willing to contribute (by handling this event). You don't need to keep the behavior verbatim. The simplest way is to change the call "set_info" to "add_info" in the communicator. This way, even if the communicator and controllers use the same time, info will be additive. But they don't need to use the same time. You could use whatever name for the communicator when adding/setting stats info.

Reply:
Reverted back to use only GET_STATS event, and changed `set_info()` to `add_info()` in the communicator so as to not overwrite. Keeping the same name for now, as this is used in other places.